test/utils/asserts.hpp

```diff
@@ -0,0 +1,18 @@
+#pragma once
+
+#include <gtest/gtest.h>
+
+#define EXPECT_THROW_WITH_MESSAGE(stmt, ex_type, expected)                                   \
```

Comment:
This is fine, but `EXPECT_THROW_WITH_MESSSAGE(...) << "hint"` will not work. The alternative solution proposed in https://github.com/google/googletest/issues/952#issuecomment-521361666 seems to fix that and is it leaves type checking to original `EXPECT_THROW`.

Reply:
I tried it, failure output is more ugly for the case of incorrect exception thrown
```
[ RUN      ] instantiate.imported_functions_not_enough
/home/andrei/dev/fizzy/test/unittests/instantiate_test.cpp:58: Failure
Expected: try { instantiate(module, {}); } catch (const std::bad_alloc& ex) { switch (0) case 0: default: if (const ::testing::AssertionResult gtest_ar = (::testing::internal::CmpHelperSTREQ("ex.what()", "\"Module requires 1 imported functions, 0 provided\"", ex.what(), "Module requires 1 imported functions, 0 provided"))) ; else ::testing::internal::AssertHelper(::testing::TestPartResult::kNonFatalFailure, "/home/andrei/dev/fizzy/test/unittests/instantiate_test.cpp", 58, gtest_ar.failure_message()) = ::testing::Message(); throw; } throws an exception of type std::bad_alloc.
  Actual: it throws a different type.
```

Reply:
For not throwing, too
```
[ RUN      ] instantiate.imported_functions_not_enough
/home/andrei/dev/fizzy/test/unittests/instantiate_test.cpp:62: Failure
Expected: try { instantiate(module1, {}); } catch (const std::runtime_error& ex) { switch (0) case 0: default: if (const ::testing::AssertionResult gtest_ar = (::testing::internal::CmpHelperSTREQ("ex.what()", "\"Module requires 1 imported functions, 0 provided\"", ex.what(), "Module requires 1 imported functions, 0 provided"))) ; else ::testing::internal::AssertHelper(::testing::TestPartResult::kNonFatalFailure, "/home/andrei/dev/fizzy/test/unittests/instantiate_test.cpp", 62, gtest_ar.failure_message()) = ::testing::Message(); throw; } throws an exception of type std::runtime_error.
  Actual: it throws nothing.
```